### PR TITLE
[TK-05043] Convert install app admin call to generate key if passed None for agent key

### PR DESCRIPTION
**Pull request by zippy**
As part of bootstrapping identity in holochain (before DeepKey and other services are completed) we need to be able to request keypair generation when calling InstallApp on the admin interface.  This is implemented in this PR by converting `agent_key` to `Option<InstallAppDnaPayload>` and using the keystore to generate the pair if `None` is specified.   Additionally this requires returning the agent pub key in the response.

Tests are updated and a new test is added for the `None` case.

### CODE REVIEW

File: crates/holochain/src/conductor/api/api_external/admin_interface.rs [hunk 1]
```diff
@@ -322,25 +344,104 @@ mod test {
...
-        assert_matches!(
-            install_response,
-            AdminResponse::AppInstalled(cell_ids) if cell_ids == expected_cell_ids
-        );
+
+        let agent_key = if let AdminResponse::AppInstalled(resp) = install_response.clone() {
+            resp.agent_key
+        } else {
+            panic!("expecting AppInstalled")
+        };
```
Comment (maackle) [inline crates/holochain/src/conductor/api/api_external/admin_interface.rs:431]: Why'd you not use `assert_matches!` here?

**Review by maackle — COMMENTED** [review]
I don't really like how this now allows two different ways to generate an agent key. If we go with this, we might as well just remove `GenerateAgentPubKey`, but it seems useful to be able to generate keys separately from the apps you're using them in. 

I'm not fully aware of the details of the bootstrapping problem that this solves, but I wonder why this is necessary, and if we shouldn't just make sure `GenerateAgentPubKey` just works in all cases.

**Review by maackle — CHANGES_REQUESTED** [review]
I want to make sure we need this first

**Comment by zippy** [comment]
@maackle I think you might be right.  Will talk with @neonphog and @artbrock

**Comment by neonphog** [comment]
I think we decided not to merge this for now. Feel free to re-open if that changes.

**Pull request closed without merging**